src/generate-query.ts

```diff
@@ -610,6 +638,170 @@ function getArgsAndVars(
   }
 }
 
+/**
+ * Given a field node, return the listSize directive, if it exists
+ */
+function getListSizeDirective(field: FieldDefinitionNode): DirectiveNode {
+  return field?.directives.find((dir) => dir.name.value === 'listSize')
+}
+
+/**
+ * Given a listSize directive, get the slicingArguments argument
+ */
+function getSlicingArguments(listSizeDirective: DirectiveNode): ArgumentNode {
+  if (!listSizeDirective) return undefined
+
+  const slicingArgumentsArg = listSizeDirective.arguments.find(
+    (arg) => arg.name.value === 'slicingArguments'
+  )
+
+  if (!slicingArgumentsArg || slicingArgumentsArg.value.kind !== 'ListValue')
+    return undefined
+
+  return slicingArgumentsArg
+}
+
+/**
+ * Given a listSize directive, get the requireOneSlicingArgument argument
+ */
+function getRequireOneSlicingArgument(
+  listSizeDirective: DirectiveNode
+): ArgumentNode {
+  if (!listSizeDirective) return undefined
+
+  const requireOneSlicingArg = listSizeDirective.arguments.find(
+    (arg) => arg.name.value === 'requireOneSlicingArgument'
+  )
+
+  if (
+    !requireOneSlicingArg ||
+    requireOneSlicingArg.value.kind !== 'BooleanValue'
+  )
+    return undefined
+
+  return requireOneSlicingArg
+}
+
+/**
+ * For a given listSize directive, return a default value for it
+ */
+function getListSizeDirectiveDefaultValue(
+  listSizeDirective: DirectiveNode,
+  typeNode: TypeNode,
+  config: InternalConfiguration,
+  schema: GraphQLSchema
+) {
+  const slicingArgumentsArg = getSlicingArguments(listSizeDirective)
+  if (!slicingArgumentsArg) return undefined
+
+  /**
+   * Already checked slicingArgumentsArguments is a list
+   *
+   * Extract paths from slicingArgumentsArguments
+   */
+  let slicingArgumentsPaths = (slicingArgumentsArg.value as ListValueNode).values.map(
+    (value) => {
+      if (value.kind === 'StringValue') return value.value
+    }
+  )
+
+  const requireOneSlicingArg = getRequireOneSlicingArgument(listSizeDirective)
+  if (
+    !requireOneSlicingArg ||
+    (requireOneSlicingArg.value as BooleanValueNode).value === true
+  ) {
+    slicingArgumentsPaths = slicingArgumentsPaths.slice(0, 1)
+  }
+  const slicingArgumentsTokenizedPaths = slicingArgumentsPaths.map((value) =>
+    value.split('.')
+  )
+
+  if (
+    slicingArgumentsPaths.length === 1 &&
+    slicingArgumentsTokenizedPaths[0].length === 1
+  )
+    return getDefaultArgValue(schema, config, typeNode)
+
+  return slicingArgumentsTokenizedPaths.reduce((res, tokenizedPath) => {
+    return merge.recursive(
+      res,
+      getListSizeDirectiveDefaultValueHelper(
+        tokenizedPath,
+        typeNode,
+        config,
+        schema
+      )
+    )
+  }, {})
+}
```

Comment:
I think this is dead code, since we either `requireOneSlicingArgument: true` so we only use the return right before or `requireOneSlicingArgument: false` so `getMissingSlicingArguments` never returns anything as "required".